modules/channel-web/src/backend/db.ts

```diff
@@ -91,27 +144,19 @@ export default class WebchatDb {
       message_raw: this.knex.json.set(raw),
       message_data: this.knex.json.set(data),
       payload: this.knex.json.set(payload),
-      sent_on: this.knex.date.now()
+      sent_on: new Date().toISOString()
     }
 
-    return Bluebird.join(
-      this.knex('web_messages')
-        .insert(message)
-        .then(),
-
-      this.knex('web_conversations')
-        .where({ id: conversationId, userId: userId, botId: botId })
-        .update({ last_heard_on: this.knex.date.now() })
-        .then(),
-
-      () => ({
-        ...message,
-        sent_on: new Date(),
-        message_raw: raw,
-        message_data: data,
-        payload: payload
-      })
-    )
+    this.queued_messages.push(message)
+    this.queued_convos[`${conversationId}_${userId}_${botId}`] = this.knex.date.now()
```

Comment:
Since one message always equates to one convo update, why not join both queries in an object instead of two separate queues ? what's the advantage?

Reply:
If a user sends 3 messages before the batch is complete, we only need to update the conversation once.